src/controllers/adminsController.js

```diff
@@ -0,0 +1,58 @@
+import { Router } from 'express';
+import HttpStatus from 'http-status-codes';
+import * as adminService from '../services/adminService';
+import { findAdmin, adminValidator } from '../validators/adminValidator';
+
+const router = Router();
+
+/**
+ * GET /api/admins
+ */
+router.get('/', (req, res, next) => {
+  adminService
+    .getAllAdmins()
+    .then(data => res.json({ data }))
+    .catch(err => next(err));
+});
+
+/**
+ * GET /api/admins/:id
+ */
+router.get('/:id', (req, res, next) => {
+  adminService
+    .getAdmin(req.params.id)
+    .then(data => res.json({ data }))
+    .catch(err => next(err));
+});
+
+/**
+ * POST /api/admin
+ */
+router.post('/', (req, res, next) => {
+  adminService
+    .createAdmin(req.body)
+    .then(data => res.status(HttpStatus.CREATED).json({ data }))
+    .catch(err => next(err));
+});
+
+/**
+ * PUT /api/admin/:id
+ */
+router.put('/:id', findAdmin, adminValidator, (req, res, next) => {
```

Comment:
What is the purpose of findAdmin middleware here ?

Reply:
to check the existence of the admin in the database, to update the admin credentials